dialogue-core/src/main/java/com/palantir/dialogue/core/BlacklistingChannel.java

```diff
@@ -50,7 +49,7 @@
     private final LimitedChannel delegate;
     private final Duration duration;
     private final Ticker ticker;
-    private final Cache<Endpoint, BlacklistState> perEndpointBlacklistState;
+    private final AtomicReference<BlacklistState> channelBlacklistState;
```

Comment:
This could be volatile since nothing takes advantage of CAS operations. I think we should be using CAS to prevent multiple failures clobbering state, but that could be done in a follow up (current semantics match the cache, just more obvious where we can improve now)